### DISCUSSION TITLE
Support for reading credentials from auth.conf

### DISCUSSION THREAD

**Discussion by SimiMilton** [rickysarraf/apt-offline#208] [Ideas]
We were trying to use apt-offline to do offline updates on Ubuntu Pro enabled systems. While trying to download the upgrades from a network connected VM(which is also pro enabled) , we are getting '401-Unauthorized' error. On further investigation, it seems that there is no support for reading credentials from auth.conf by apt-offline. The authentication required for accessing the apt repositories is stored in `/etc/apt/auth.conf.,d/90ubuntu-upgrade`. This file doesn't appear to be used by apt-offline when accessing these repositories. 

Does anyone know if the support is already there in any of the existing forks of apt-offline ?

**Comment by rickysarraf** [comment]
If it is just HTTP Basic Auth, you may make use of the `--http-basicauth` option and pass a properly crafted URL. I haven't used that option myself but it was contributed by an apt-offline user and should work.

```
       --proxy-host
                 Specify the Proxy Host to be used.

       --proxy-port
                 Specify the Proxy port number to be used.

       --https-cert-file FILENAME
                 The Certificate-file to use if https-client-authentication is required on one or multiple hosts.

       --https-key-file FILENAME
                 The Certificate-key-file to use if https-client-authentication is required on one or multiple hosts.

       --disable-cert-check
                 Disable the check of https certificates on servers

       --http-basicauth URL
                 A username/password encoded in the URL. Note Passwords with special characters should be percent encoded as per RFC-3986 section 2.1
```





>The authentication required for accessing the apt repositories is stored in /etc/apt/auth.conf.,d/90ubuntu-upgrade. This file doesn't appear to be used by apt-offline when accessing these repositories.

That is expected behavior. Those configuration bits are specific to the `apt` tool and not `apt-offline`.  `apt-offline` does not depend on `apt` for its `get` operation.

**Reply by SimiMilton** [reply]
Thanks for you response. We are using signature files generated in an offline VM for downloading the updates via a network connected VM. What you proposed won't work in that case?
Edit: Sorry, I think I overlooked your comment, --http-basicauth does work with signature files. Thanks again!

**Reply by rickysarraf** [reply]
Glad to know all sorted out.

**Reply by RahulJadhav26** [reply]
Can you suggest how to pass the basic auth for my private repo. For example if my user name is rahul and password is password#123
Even I pass my URL https://rahul:password%23123@host.com. It seems that verbose says that added user rahul with password password#123 to host.com but it still doesnt work.

**Reply by rickysarraf** [reply]
@RahulJadhav26 `It doesn't work` is not helpful. Do you get the same error as  was reported initially ?
